toolchain/lower/testdata/interop/cpp/thunk.carbon

```diff
@@ -0,0 +1,77 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/int.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/lower/testdata/interop/cpp/thunk.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/lower/testdata/interop/cpp/thunk.carbon
+
+// ============================================================================
+// Parameter requires a thunk to be generated
+// ============================================================================
+
+// --- thunk.h
+
+void foo(short x);
+
+// --- import_thunk.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "thunk.h";
+
+fn MyF() {
+  Cpp.foo(5);
+}
+
+// CHECK:STDOUT: ; ModuleID = 'import_thunk.carbon'
+// CHECK:STDOUT: source_filename = "import_thunk.carbon"
+// CHECK:STDOUT: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128"
+// CHECK:STDOUT: target triple = "x86_64-unknown-linux-gnu"
+// CHECK:STDOUT:
+// CHECK:STDOUT: define void @_CMyF.Main() !dbg !7 {
+// CHECK:STDOUT: entry:
+// CHECK:STDOUT:   %.loc7_12.1.temp = alloca i16, align 2, !dbg !10
```

Comment:
This isn't working -- we never store the value 5 into this `alloca`. This is a result of the direct use of an `initialize_from` when building the thunk call rather than calling into the full `Initialize` machinery, though, so if you want to consider that to be addressed by the TODO in `PerformCppThunkCall` I'd be OK with that -- though adding a TODO comment to this test too would be nice.

Reply:
Added TODOs.